src/net_processing.cpp

```diff
@@ -2383,6 +2401,31 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
                 }
             }
         }
+        // If we're in IBD, we want outbound peers that will serve us a useful
```

Comment:
Seems like all the timeout checking could be pulled out into a separate function.